File: Programmers/pro_49993_스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    def key(x):
        return skill.index(x)
    for i in skill_trees:
        cmp_trees = [x for x in i if x in skill]
        cmp_str=''.join(cmp_trees)
        if(cmp_str == ''.join(sorted(cmp_trees,key=lambda x:key(x)))):
            if(skill[:len(cmp_trees)]==cmp_str):
                answer+=1
    return answer

File: Programmers/pro_49993_스킬트리.js
function solution(skill, skill_trees) {
    const len = skill_trees.length
    var answer = 0;
    for (let i = 0; i < len; i++) {
        let filter_skill = [...skill_trees[i]].filter((v) => skill.indexOf(v) != -1).join("")
        if (skill.slice(0, filter_skill.length) === filter_skill)
            answer += 1
    }
    return answer;
}